src/main/kotlin/com/learnspigot/bot/profile/ProfileCommand.kt

```diff
@@ -21,21 +24,58 @@ class ProfileCommand {
     )
     fun onProfileCommand(
         event: SlashCommandInteractionEvent,
-        @Description("User to show profile") @Optional user: User?
+        @Description("User to show profile") @Optional user: User?,
+        @Description("URL to show profile") @Optional url: String?
     ) {
-        val finalUser = user ?: event.user
-        val profile = profileRegistry.findByUser(finalUser)
-
-        event.replyEmbeds(
-            embed()
-                .setTitle("Profile Lookup")
-                .addField("Discord", finalUser.name + " (" + finalUser.asMention + ")", false)
-                .addField("Udemy", profile.udemyProfileUrl ?: "Not linked", false)
-                .addField("Reputation", profile.reputation.size.toString(), true)
-                .addField("(Notifications)", profile.notifyOnRep.toString(), true)
-                .setThumbnail(finalUser.effectiveAvatarUrl)
-                .build()
-        ).setEphemeral(true).queue()
+        val profileByURL = profileRegistry.findByURL(url ?: "")
+
+        val embed: MessageEmbed = when {
+            user == null && url == null -> {
+                userProfileEmbed(event.user)
+            }
+            user != null && url != null -> {
+                embed()
```

Comment:
We are constantly repeating the process of creating the same embed, we can move this logic out into a function, something like "createProfileLookupEmbed", e.g.
```kt
fun createProfileLookupEmbed(title: String, description: String): MessageEmbed = embed()
    .setTitle("Profile Lookup")
    .addField(title, description, false)
    .build()
```
We can then use this everywhere we need it, and we can specify a title and description to make our when less "heavy". F.x.
```kt
user != null && url != null -> createProfileLookupEmbed("Please choose", "Please make a choice whether you want to use a user or use a URL.")
```

Reply:
By taking this out into a function, we can actually make all of these conditions 1 liners